components/common/listingCart/ListingCartModal.vue

```diff
@@ -0,0 +1,210 @@
+          {{ $t('listingCart.potentialEarnings') }}
+          <div class="is-flex">
+            <span class="ml-2 has-text-grey"
+              >{{ totalNFTsPrice.toFixed(4) }} {{ chainSymbol }}</span
+            >
+            <span class="has-text-weight-bold ml-2"> ${{ priceUSD }} </span>
+          </div>
+        </div>
+
+        <div class="is-flex is-justify-content-space-between pb-5 px-6">
+          <NeoButton
+            :disabled="Boolean(listingCartStore.incompleteListPrices)"
+            :label="confirmListingLabel"
+            variant="k-accent"
+            no-shadow
+            class="is-flex is-flex-grow-1 py-5"
+            @click.native="confirm" />
+        </div>
+      </div>
+    </NeoModal>
+  </div>
+</template>
+<script setup lang="ts">
+import { Interaction } from '@kodadot1/minimark/v1'
+import ListingCartPriceInput from '@/components/common/listingCart/ListingCartPriceInput.vue'
+import { prefixToToken } from '@/components/common/shoppingCart/utils'
+import IdentityItem from '@/components/identity/IdentityItem.vue'
+import { NeoButton, NeoModal } from '@kodadot1/brick'
+import { usePreferencesStore } from '@/stores/preferences'
+import { TokenToList } from '@/composables/transaction/types'
+import { useListingCartStore } from '@/stores/listingCart'
+import { calculateBalance } from '@/utils/format/balance'
+import { warningMessage } from '@/utils/notification'
+import { useFiatStore } from '@/stores/fiat'
+import { calculateExactUsdFromToken } from '@/utils/calculation'
+const { isLogIn, accountId } = useAuth()
+const { urlPrefix } = usePrefix()
+const preferencesStore = usePreferencesStore()
+const listingCartStore = useListingCartStore()
+const { transaction, isLoading, status } = useTransaction()
+const { $i18n } = useNuxtApp()
+
+const { chainSymbol } = useChain()
+const fixedPrice = ref<number | null>(null)
+const floorPricePercentAdjustment = ref(1)
+function setFixedPrice() {
+  listingCartStore.setFixedPrice(fixedPrice.value)
+  fixedPrice.value = null
+}
+const fiatStore = useFiatStore()
+const priceUSD = computed(() =>
+  calculateExactUsdFromToken(
+    totalNFTsPrice.value,
+    Number(fiatStore.getCurrentTokenValue(prefixToToken[urlPrefix.value]))
+  )
+)
+
+const totalNFTsPrice = computed(() =>
+  listingCartStore.itemsInChain.reduce((acc, nft) => {
+    return Number((acc + Number(nft.listPrice || 0)).toFixed(4))
+  }, 0)
+)
+
+const confirmListingLabel = computed(() => {
+  switch (listingCartStore.incompleteListPrices) {
+    case 0:
+      return $i18n.t('listingCart.complete')
+    case 1:
+      return $i18n.t('listingCart.missing1')
+    default:
+      return `${listingCartStore.incompleteListPrices} ${$i18n.t(
+        'listingCart.missingMultiple'
+      )}`
+  }
+})
+async function confirm() {
+  const token = listingCartStore.itemsInChain
+    .filter((item) => item.listPrice)
+    .map((item) => ({
+      price: String(calculateBalance(item.listPrice ?? 0)),
```

Comment:
fallback isn't really necessary here even with the original 
<img width="244" alt="image" src="https://github.com/kodadot/nft-gallery/assets/3810177/b257a6e7-40ad-4e06-8a99-ac88f3276bec">

But typescript don't know that we already filtered it thus throwing some errors. So placed a fallback to silence it.

<img width="1082" alt="image" src="https://github.com/kodadot/nft-gallery/assets/3810177/e0612d8c-c376-466b-b932-e7fb8584a259">

I already change that to Boolean now for better readability. Still TS error here, please advice what would you want me de

Reply:
I just did a type assertion to silence it instead of ?? 0
<img width="810" alt="image" src="https://github.com/kodadot/nft-gallery/assets/3810177/3b88243c-a362-477b-8d38-d8b1eeb3bf7f">